flashinfer/prefill.py

```
@@ -3853,3 +3850,84 @@ def fmha_v2_prefill_deepseek(
+        Output tensor, or (output, lse) if return_lse is True.
+    """
+    if not (is_sm120a_supported(query.device) or is_sm121a_supported(query.device)):
+        raise ValueError(
+            "fmha_v2_prefill_sm120 is only supported on SM12x GPUs "
+            "(RTX 5090, DGX Spark GB10)."
+        )
+    head_dim = query.shape[3]
+    if head_dim not in (64, 128):
+        raise ValueError(
+            f"fmha_v2_prefill_sm120 only supports head_dim 64 or 128, got {head_dim}."
+        )
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)